src/node/internal/internal_net.ts

```diff
@@ -1753,3 +1749,529 @@ export class SocketAddress implements _SocketAddress {
     return undefined;
   }
 }
+
+// ======================================================================================
+
+// Note: the bulk of the following code was authored by claude...
+// The implementation in Node.js is split between javascript and C++.
+// Here we do the entire implementation as TypeScript simply because
+// we don't need the C++ parts at all.
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+// IPv4/IPv6 CIDR network utilities
+function parseIPv4(ip: string): number[] {
+  const parts = ip.split('.');
+  if (parts.length !== 4) return [];
+
+  const nums = parts.map((p) => {
+    const num = parseInt(p, 10);
+    return num >= 0 && num <= 255 ? num : -1;
+  });
```

Comment:
nits:
- could this be a for loop with an early return?
- maybe have 2 early returns for clarity: `Number.isNaN()` and out of bounds (0-255)